conferences/ticketing_storage.go

```diff
@@ -0,0 +1,605 @@
+
+	return &results, nil
+}
+
+func readConferenceSlotByID(ctx context.Context, tx *sqldb.Tx, id uint64, loadDeps bool) (*ConferenceSlot, error) {
+	results := ConferenceSlot{}
+	row := queryRow(ctx, tx,
+		`SELECT (id, name, descripcion, cost, capacity, start_date, end_date, purchaseable_form, purchaseable_until, available_to_public, conference_id, depends_on_id)
+	FROM conference_slot 
+	WHERE id = $1`, id)
+
+	var dependsOnID uint64
+	var conferenceID uint64
+
+	err := row.Scan(&results.ID,
+		&results.Name,
+		&results.Description,
+		&results.Cost,
+		&results.Capacity,
+		&results.StartDate,
+		&results.EndDate,
+		&results.PurchaseableFrom,
+		&results.PurchaseableUntil,
+		&results.AvailableToPublic,
+		&conferenceID,
+		&dependsOnID)
+	if err == sql.ErrNoRows {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("reading conference slots by id: %w", err)
+	}
+
+	/*
+			if dependsOnID != 0 && loadDeps {
+				results.DependsOn, err = readConferenceSlotByID(ctx, tx, dependsOnID, false)
+				if err != nil {
+					return nil, fmt.Errorf("loading dependency: %w", err)
+				}
+			}
+
+
+		conference := Conference{}
+		row = queryRow(ctx, tx,
+			`SELECT id, name, slug, start_date, end_date, location FROM conference WHERE id = $1`, conferenceID)
+		err = row.Scan(&conference.ID, &conference.Name, &conference.Slug, &conference.StartDate, &conference.EndDate, &conference.Location)
+		if err != nil {
+			return nil, fmt.Errorf("reading conference by id: %w", err)
+		}
+
+		results.Conference = &conference
+	*/
+	return &results, nil
+}
+
+// updateConferenceSlot updates conference slot fields from the passed instance
+func updateConferenceSlot(ctx context.Context, tx *sqldb.Tx, cslot *ConferenceSlot, conferenceID int64) error {
+	var sqlStatement = `UPDATE conference_slot 
+	SET conference_id = $1, name = $2, descripcion = $3, cost =$4, capacity=$5, 
+	start_date = $6, end_date = $7, purchaseable_form = $8, purchaseable_until = $9, 
+	available_to_public $10, depends_on_id = NULL
+	WHERE id = $11`
+	var args = []interface{}{
+		conferenceID,
+		cslot.Name,
+		cslot.Description,
+		cslot.Cost,
+		cslot.Capacity,
+		cslot.StartDate,
+		cslot.EndDate,
+		cslot.PurchaseableFrom,
+		cslot.PurchaseableUntil,
+		cslot.AvailableToPublic,
+	}
+	/* if e.DependsOn != nil {
+		sqlStatement = `UPDATE conference_slot
+	SET conference_id = $1, name = $2, descripcion = $3, cost =$4, capacity=$5,
+	start_date = $6, end_date = $7, purchaseable_form = $8, purchaseable_until = $9,
+	available_to_public $10, depends_on_id = $11
+	WHERE id = $12`
+		args = append(args, e.DependsOn.ID)
+	} */
+	args = append(args, cslot.ID)
+
+	res, err := exec(ctx, tx, sqlStatement, args...)
+
+	if err != nil {
+		return fmt.Errorf("updating conference slot: %w", err)
+	}
+	ra, err := res.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("updting conference slot: %w", err)
+	}
+	if ra == 0 {
+		return fmt.Errorf("slot was not updated")
```

Comment:
Since we are using a `WHERE` this error would be due to there not being a slot, rather than not updating a slot. 
```suggestion
		return fmt.Errorf("no such slot")
```